modules/ppcp-agentic-commerce/src/Endpoint/AgenticRestEndpoint.php

```diff
@@ -155,7 +153,14 @@ protected function parse_and_validate_cart( WP_REST_Request $request ) {
 
 		$issues = $cart->validate();
 		if ( ! empty( $issues ) ) {
-			return new BadRequestError( 'Cart validation issue', $issues );
+			$issue_details = array_map(
+				function ( $issue ) {
+					return $issue->to_array();
+				},
+				$issues
+			);
+
+			return new BadRequestError( 'Cart validation issue', $issue_details );
```

Comment:
This is actually wrong logic - validation issues should never trigger an error.

Error = we cannot process the request (authentication, empty/malformed payload, ...)
Validation = we processed the data but it's not valid for checkout